core/build.gradle

```
@@ -205,24 +188,55 @@ dependencies {
-    testCompile 'com.rabbitmq:amqp-client:5.9.0'
-    testCompile 'org.mongodb:mongo-java-driver:3.12.7'
+    testImplementation 'org.apache.httpcomponents:httpclient:4.5.9'
+    testImplementation 'redis.clients:jedis:3.6.0'
+    testImplementation 'com.rabbitmq:amqp-client:5.9.0'
+    testImplementation 'org.mongodb:mongo-java-driver:3.12.7'
 
-    testCompile ('org.mockito:mockito-core:3.10.0') {
+    testImplementation ('org.mockito:mockito-core:3.10.0') {
         exclude(module: 'hamcrest-core')
     }
     // Synthetic JAR used for MountableFileTest and DirectoryTarResourceTest
-    testCompile files('testlib/repo/fakejar/fakejar/0/fakejar-0.jar')
+    testImplementation files('testlib/repo/fakejar/fakejar/0/fakejar-0.jar')
 
-    testCompile 'org.rnorth.visible-assertions:visible-assertions:2.1.2'
-    testCompile 'org.assertj:assertj-core:3.18.1'
-    testCompile project(':test-support')
+    testImplementation 'org.rnorth.visible-assertions:visible-assertions:2.1.2'
+    testImplementation 'org.assertj:assertj-core:3.18.1'
+    testImplementation project(':test-support')
 
     jarFileTestCompileOnly "org.projectlombok:lombok:${lombok.version}"
     jarFileTestAnnotationProcessor "org.projectlombok:lombok:${lombok.version}"
-    jarFileTestCompile 'junit:junit:4.12'
-    jarFileTestCompile 'org.assertj:assertj-core:3.18.1'
-    jarFileTestCompile 'org.ow2.asm:asm-debug-all:5.2'
+    jarFileTestImplementation 'junit:junit:4.12'
+    jarFileTestImplementation 'org.assertj:assertj-core:3.18.1'
+    jarFileTestImplementation 'org.ow2.asm:asm-debug-all:5.2'
 }
+
+tasks.generatePomFileForMavenJavaPublication.finalizedBy(
+    // The task is a bit more generic that it needs to be, but we may consider applying it to other modules too
+    tasks.register('checkPOMdependencies') {
+        doLast {
+            def rootNode = new XmlSlurper().parse(tasks.generatePomFileForMavenJavaPublication.destination)
+
+            def artifactId = rootNode.artifactId.text()
+
+            def latestRelease = new XmlSlurper()
+                .parse("https://repo1.maven.org/maven2/org/testcontainers/${artifactId}/maven-metadata.xml")
+                .versioning.release.text()
+
+            def releasedRootNode = new XmlSlurper()
+                .parse("https://repo1.maven.org/maven2/org/testcontainers/${artifactId}/${latestRelease}/${artifactId}-${latestRelease}.pom")
+
+            Set<String> dependencies = releasedRootNode.dependencies.children()
+                .collect { "${it.groupId.text()}:${it.artifactId.text()}".toString() }
+
+            Set<String> whitelist = [
+            ]
+
+            for (dependency in rootNode.dependencies.children()) {
+                def coordinates = "${dependency.groupId.text()}:${dependency.artifactId.text()}".toString()
+                if (!dependencies.contains(coordinates) && !whitelist.contains(coordinates)) {
+                    throw new IllegalStateException("New dependency! ${coordinates}")
```

Could we make this exception message more helpful for contributors?

e.g.
```suggestion
                    throw new IllegalStateException("A new dependency has been added ${coordinates} - if this was intentional please add it to the whitelist in PATH_TO_GRADLE_FILE")
```

What do we expect the process to be - do we need to be approving new dependencies somehow?

BTW I think it would be good to move this to a separate `.gradle` file, so that areas of major scripting are encapsulated.

---

Thanks for the suggestion! We could, yes 👍 
FYI this message is mostly for us, maintainers, as the chance of an external dependency added by external contributors is super low

---

Extracted and made the message more user friendly 👍